### PR TITLE
drop requests if client disconnects

### PR DESCRIPTION
**Pull request by drmorr0**
I ran @stug's load test in PAASTA-15933 and it _seems_ that things are better with this change.  This is the output from before the change is applied: 
https://fluffy.yelpcorp.com/i/bmFVkxnBCbjCT5bszfxwkvF9dG8nHJZ5.html

And this is the output from after:
~https://fluffy.yelpcorp.com/i/9H7KjqzWkB96g0LflL5tJPXMpNfVBBrg.html~
https://fluffy.yelpcorp.com/i/gz0d1b1BZzDxMDX0Dq7TmjQ6cPH1BGmS.html

I think this means that the queue isn't getting saturated "as much"?  But it's a little hard for me to tell.

### CODE REVIEW

File: hacheck/handlers.py [hunk 1]
```diff
@@ -144,10 +145,17 @@ def get(self, service_name, port, query):
...
-                self.write(last_message)
-                self.finish()
+                if not self.timed_out:
+                    last_statuses[service_name] = StatusResponse(200, self.request.remote_ip, time.time())
+                    self.set_status(200)
+                    self.write(last_message)
+                    self.finish()
+
+    def on_connection_close(self):
+        self.timed_out = True
```
Comment (stug) [inline hacheck/handlers.py]: are you sure that this doesn't persist across requests somehow?  seems like it shouldn't, but just want to make sure.

File: hacheck/handlers.py [hunk 2]
```diff
@@ -144,10 +145,17 @@ def get(self, service_name, port, query):
...
+        self.set_status(504)
+        self.write('Upstream client closed the connection')
+        self.finish()
```
Comment (stug) [inline hacheck/handlers.py]: are you positive that this actually closes the downstream healthcheck request?

File: hacheck/handlers.py (hunk 1 above)
Comment (drmorr0) [reply hacheck/handlers.py]: Discussed on slack; this gets reset at the start of `get()`

**Comment by drmorr0** [comment]
Ok, I just updated the code to cancel the future as discussed on Slack.  It's still not clear to me from the sample output that this is doing any good?

**Comment by drmorr0** [comment]
We are discarding this PR because it doesn't actually help.

**Pull request closed without merging**